cachi2/core/package_managers/pip.py

```diff
@@ -1,4 +1,3 @@
-# SPDX-License-Identifier: GPL-3.0-or-later
```

Comment:
Why do you remove license identifier?

Reply:
That is a leftover from Cachito.

Reply:
It should stay IMO. It's always harder to add SPDX license IDs afterwards should we adopt modules which would simply not comply with GPL, but would with LGPL (e.g. contributions from entities working primarily closed source). IOW should it happen that we wouldn't have a global license, these SPDX tags very clearly denote what module is shipped under what license. They're harmless and I'd argue we should add them to all modules if we haven't done so yet while we can (i.e. while the contributions are coming from the same entity).

Reply:
Back in the day every module in OpenStack would contain several lines of header to indicate which license it is distributed under. The reasoning was that if any of the modules start living on their own there would be no confusion about how they must be licensed. Back then the general consensus was that some hints about governing license must be present everywhere.

Reply:
I don't know anything about this topic so I will add it back. Btw. there are only a few files that contain these license identifiers.

Reply:
> Back in the day every module in OpenStack would contain several lines of header to indicate which license it is distributed under. The reasoning was that if any of the modules start living on their own there would be no confusion about how they must be licensed. Back then the general consensus was that some hints about governing license must be present everywhere.

Yes. A bit of trivia on a different project:

We had a bit different situation in libvirt - the project is distributed under both LGPL and GPL (the library code was LGPL and library unrelated code was GPL). Another major factor there were copyrighted contributions all of which predated SPDX license tags. Since the project has matured and has been around for a while there was only a limited maneuvering space to do something about this and so some modules had to retain the full blown LGPL preamble at the beginning mostly due to copyright mentions while new modules only use SDPX tags. In some cases the adoption of SPDX tags (in a sibling project under the same project umbrella I think) required contacting individual contributors to get their blessing for ridding of the copyright parts in the modules' preambles.

Arguably very few engineers actually understand all the nooks and crannies of OSS licensing (I'm definitely not one of them, quite the opposite!). Our situation is very different and much easier because we're operating under the global GPL license and we don't carry any copyrighted contributions. That said, removing SPDX license identifiers is rarely a good idea and we MUST be very careful with that. With all that said we have 2 options:

1. in order to be consistent, we should add the SPDX license IDs to all modules since thanks to the global license we don't need to go and ask anyone for permission to add these tags to all modules, but asking every contributor to add them to their newly created modules may sound tedious
2. Do nothing and keep it as is! Modules carried over from another project must keep whatever licensing burden they originally had and new ones are covered by the global license.

Having thought about it overnight, I have to retract on my earlier comment:

> I'd argue we should add them to all modules if we haven't done so yet while we can

and express my clear preference for 2.

Tangentially related:
As a consumer of a project one naturally cannot "downgrade" the license, only the opposite direction is possible. However, I wonder what the steps for the project itself are if they wish to re-license, downgrading to LPGL for example, it surely would require a unilateral vote of the community (yeah, good luck with that), but I don't know what the process actually looks like at all.